src/components/Header/NavComponents/userprofile.tsx

```diff
@@ -30,6 +41,73 @@ export const UserProfile: React.FC<{}> = () => {
   const { data: agaveBalance } = useAmountAvailableToStake(address);
   const userBal = agaveBalance ? bigNumberToString(agaveBalance, 3) : "0";
 
+  const chainAddresses = useChainAddresses();
+
+  // internalAddressesPerNetworkId cannot be indexed for each chain, strigified and parsed back to index
+  var chains =  JSON.parse(JSON.stringify(internalAddressesPerNetworkId))
+
+  // Buttons to change to every available chain
+  let buttons : any[] = [];
+  Object.entries(internalAddressesPerNetworkId).forEach(([name, chain]) => {
+    buttons.push(
+      <Button 
+        bg={mode({ base: "primary.500", md: "primary.500" }, "primary.500")}
+        colorScheme="teal"
+        size="xl"
+        h="40px"
+        onClick={() => changeId(chain.chainId)}
+      >
+        {chain.chainName}
+      </Button>
+  )})
+
+  const currChainName : String = chains[chainAddresses?.chainId || 0]?.chainName
+
+  // Request wallet provider to change chain
+  // TODO: change it to enother file
+  function changeId(chainId : Number) {
```

Comment:
`number`, not `Number`; the latter is a specialized wrapper type and cannot / should not be instantiated.
```suggestion
  function changeId(chainId: number) {
```

Reply:
Fixed